src/lib/protocols/dns.c

```diff
@@ -513,6 +517,26 @@ static void ndpi_search_dns(struct ndpi_detection_module_struct *ndpi_struct, st
 
   if(flow->packet_counter > 3)
     NDPI_EXCLUDE_PROTO(ndpi_struct, flow);
+    
+    
+  if(flow->packet.udp != NULL && flow->packet.payload_packet_len > PKT_LEN_ALERT)
```

Comment:
1) Are you sure this limit applies to  LLMNR and MDNS, too (other than "standard" DNS)?
2) I don't know if it is in scope with your project, but it is more complex than that even for "standard" DNS: RFC 6891 allows DNS packets to be bigger than 512 bytes...

Reply:
Ciao, grazie per il tuo feedback!
Questa pull request è un progetto che ho fatto per il corso Gestione di Reti con il professor Luca Deri, quindi i due flow risk sono stati implementati con il suo accordo. Ho iniziato con queste informazioni che indicano i problemi che possono derivare da queste condizioni non verificate: #1282 Penso che questo risponda al punto n. 2.

Per il punto n. 1 invece sono d'accordo che potrebbe essermi sfuggito qualcosa nell'implementazione, mi informo su quanto mi hai indicato. Se avessi dei suggerimenti/riferimenti su cui posso partire andrebbe benissimo.

Grazie,
Debora

Reply:
1) rfc4795 sec 2.1 and rfc6762 sec 17
2) I was right to think it was out-of-scope for your project... But I think that you should, at least, add a comment about EDNS and its future implementation to avoid false positives